### PR TITLE
Controller topology

### PR DESCRIPTION
**Pull request by KOPRajs**
This PR adds:
- 3 button controller
- 4 players support
- New icon

The 3 button controller lets you use the new default button map (X, A, B instead of A, B, R1).

To use more than 2 players you need to set the _Teamplayer_ or _4wayplay_ in the advanced emulator settings as _Input 1_. I've looked through the Picodrive's libretro frontend source code and it seems that 4 players are maximum that is supported.

The icon is one that I've made for my personal use several years back when there was no icon for this add-on at all. Unlike _Genesis Plus GX_ the _Picodrive_ has the ability to emulate Sega 32X system, hence the picture shows the console with the 32X add-on.

### CODE REVIEW

**Comment by garbear** [comment]
Input looks great, thanks!

I like the new icon too. Is there any way you can make the dimensions 512x512?

**Comment by KOPRajs** [comment]
I was lucky enough to find the original GIMP source file, so I've exported it in 512x512 (the original is 540x540 so we can't get much more). I remember I was exporting it in this resolution to match the _Genesis Plus GX_ icon.

Here is another version (without the shadow for now):
icon

If you prefer this one I can create the shadow and replace it (the source image for this one is much better in terms of quality).

**Comment by KOPRajs** [comment]
Well, I couldn't resist. So here are both versions side by side, pick whichever you like more.
icon
icon

**Comment by garbear** [comment]
On a white background I like the shadow more muted, but we can't compare that against a dark background. From a content perspective the icon with the drive shows more gaming history than the icon without. So I think I'll have to go with option 1.

**Comment by KOPRajs** [comment]
I agree that from the content perspective the image with the drive shows more. On the other hand I was unable to find better image with both the drive and the _32X_ add-on and even the logo itself is more sharp on the other image. Also the _Sega CD_ (with the drive) is preferred to be played in the _Genesis Plus GX_, only the _Sega 32X_ is where users should go with the _Picodrive_, so maybe the second image feels better for me.

I've updated the image again (this time with the transparent background). If you agree, it is ready to be merged. If you still prefer the first image, I will swap them one more time, no problem.

You can compare the final versions here:
https://github.com/kodi-game/game.libretro.picodrive/compare/79017bdf787cb9c741434f84573b2b408b6d2289..60b24ea422b1e08213cf981d31968f51d33e0d91

**Comment by garbear** [comment]
After reviewing your comparisons I completely agree with the current state of the PR. Will merge in 24 hrs if no objections

File: game.libretro.picodrive/addon.xml.in [hunk 1]
```diff
@@ -1,10 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <addon id="game.libretro.picodrive"
 		name="Sega - MS/GG/MD/CD/32X (PicoDrive)"
-		version="2.00.0.40"
+		version="2.00.0.42"
```
Comment (garbear) [inline game.libretro.picodrive/addon.xml.in]: Can you revert? The CI was handling one version bump just fine but was breaking on two version bumps.

Comment (KOPRajs) [reply game.libretro.picodrive/addon.xml.in]: I've added a third commit to fix the version number. Is it ok? In the future, should I avoid bumping the version completely, or is the bump by 1 ok?

Comment (garbear) [reply game.libretro.picodrive/addon.xml.in]: Avoid bumping the version, best to let the CI script take care of it.

**Pull request merged**